src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -52,7 +58,20 @@ internal abstract partial class PatternMatcher : IDisposable
             bool allowFuzzyMatching = false)
         {
             culture = culture ?? CultureInfo.CurrentCulture;
+            if (culture.Name == "en-US")
```

Comment:
> I find it highly questionable that people would write code that needs culture sensitive results in en-us. And even if they did, i'm ok with that scenario regressing given the gain here.

I do not disagree with this. However, I am still generally concerned with any code that singles out en-US. It's a bad precedent that puts us in to a "works on my machine" state when interpreting feedback.  I checked the VS app domain manager code that sets CurrentUICulture based on configured VS language and, AFAICT, it leaves CurrentCulture alone. This means that this optimization (as currently written) will miss a lot of English VS users outside the US. We could tweak it to detect if the VS language is English, but then we'd still miss a not uncommon case of non-English VS with ASCII-only patterns and data.


> At worst the code performs the same as today for non-en-us. But for the common case of en-us and english code, we get a boost.

en-US is "common", but nowhere near "99.9%" mentioned elsewhere. I will forward you some data by email. 

I'd like to discuss other ways to achieve this optimization for a greater number of users. Here's my quick brainstorm on that:

* You mentioned elsewhere that there is an index that is built for the search. Is it possible to determine if the data is all ASCII during indexing and then use Ordinal only when that is the case and the pattern is all ASCII?

* Could we make ordinal vs culture-sensitive search an option that can be toggled by the user irrespective of their locale or VS language? Make Ordinal the default?

* Could we optimistically use ordinal and only fall back to culture-sensitive when no matches are found?